src/bit.rs

```diff
@@ -0,0 +1,169 @@
+/// Yields the bits of a `u64`.
+pub struct BitIterator {
+    /// The value that is being iterated bit-wise
+    value: u64,
+    /// True bits in the `mask` are the bits that have been visited.
+    mask: u64,
+}
+
+impl BitIterator {
+    pub fn new(value: u64) -> BitIterator {
+        BitIterator { value, mask: 0 }
+    }
+
+    /// An efficient skip implementation.
+    ///
+    /// Note: The compiler is smart enough to translate a `skip(n)` into a single shift instruction
+    /// if the code is inlined, however inlining does not always happen.
+    pub fn skip_front(mut self, n: u32) -> Self {
+        let mask = bitmask(n);
+        let ones = self.mask.trailing_ones();
+        let mask_position = ones;
+        self.mask ^= mask << mask_position;
+        self
+    }
+
+    /// An efficient skip from the back.
+    ///
+    /// Note: The compiler is smart enough to translate a `skip(n)` into a single shift instruction
+    /// if the code is inlined, however inlining does not always happen.
```

Comment:
nit: `skip(n)` should probably be changed to `skip_back(n)`.

Reply:
[`skip`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.skip) refers to the stdlib's method